dotCMS/src/main/java/com/dotcms/api/system/event/local/LocalSystemEventsAPIImpl.java

```diff
@@ -0,0 +1,205 @@
+package com.dotcms.api.system.event.local;
+
+import com.dotcms.concurrent.DotConcurrentFactory;
+import com.dotcms.concurrent.DotSubmitter;
+import com.dotmarketing.util.Logger;
+import com.google.common.annotations.VisibleForTesting;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * Default implementation.
+ * The implementation will be provided by {@link LocalSystemEventsAPIFactory}
+ * @author
+ */
+class LocalSystemEventsAPIImpl implements LocalSystemEventsAPI, Serializable {
+
+    public static final String LOCAL_SYSTEM_EVENTS_THREAD_POOL_SUBMITTER_NAME = "localsystemevents.submmiter";
+
+    private final ConcurrentHashMap<Class<?>, CopyOnWriteArrayList<EventSubscriber>>  eventSubscriberByEventTypeMap;
+    private final DotSubmitter dotSubmitter;
+    private final EventSubscriberFinder eventSubscriberFinder;
+    private volatile EventSubscriber<OrphanEvent> orphanEventSubscriber;
+
+    public LocalSystemEventsAPIImpl () {
+
+        this(new ConcurrentHashMap<>(), DotConcurrentFactory.getInstance(),
+                new SubscriberAnnotationEventSubscriberFinder(),
+                new DefaultOrphanEventSubscriber());
+    }
+
+    @VisibleForTesting
+    public LocalSystemEventsAPIImpl(final ConcurrentHashMap<Class<?>, CopyOnWriteArrayList<EventSubscriber>> eventSubscriberByEventTypeMap,
+                                    final DotConcurrentFactory dotConcurrentFactory,
+                                    final EventSubscriberFinder eventSubscriberFinder,
+                                    final EventSubscriber<OrphanEvent> orphanEventSubscriber) {
+
+        this.eventSubscriberByEventTypeMap = eventSubscriberByEventTypeMap;
+        this.dotSubmitter                  = dotConcurrentFactory.getSubmitter(LOCAL_SYSTEM_EVENTS_THREAD_POOL_SUBMITTER_NAME);
+        this.eventSubscriberFinder         = eventSubscriberFinder;
+        this.orphanEventSubscriber         = orphanEventSubscriber;
+    }
+
+    @Override
+    public void subscribe(final Object subscriber) {
+
+        final  Map<Class<?>, EventSubscriber> subscriberMap =
+                this.eventSubscriberFinder.findSubscribers(subscriber);
+
+        if (null != subscriberMap && !subscriberMap.isEmpty()) {
+
+            subscriberMap.forEach((eventType, eventSubscriber) -> this.subscribe(eventType, eventSubscriber));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)